src/api/MetadataQuery.js

```diff
@@ -98,32 +123,34 @@ class MetadataQuery extends Base {
      * @return {Object} - flattened metadata entry without the $ fields
      */
     flattenMetadata = (metadata: MetadataQueryResponseEntryMetadata): FlattenedMetadataQueryResponseEntryMetadata => {
-        let flattenedMetadata = {};
-
-        Object.keys(metadata).forEach(scope => {
-            Object.keys(metadata[scope]).forEach(templateKey => {
-                const nonconformingInstance = metadata[scope][templateKey];
-                const data = omit(nonconformingInstance, [
-                    '$id',
-                    '$parent',
-                    '$type',
-                    '$typeScope',
-                    '$typeVersion',
-                    '$version',
-                ]);
-
-                flattenedMetadata = {
-                    data,
-                    id: nonconformingInstance.$id,
-                    metadataTemplate: {
-                        type: 'metadata-template',
-                        templateKey,
-                    },
+        const templateFields = getProp(this.templateInfo, 'fields', []);
+        const instance = getProp(metadata, `${this.templateScope}.${this.templateKey}`);
+
+        if (!instance) {
+            return {};
+        }
+
+        const fields = Object.keys(instance)
+            .filter(key => !key.startsWith('$'))
```

Comment:
Previously, this code filtered out only six keys. Do we want to filter out all keys that start with `$`?

Reply:
Yes